tmt/base.py

```diff
@@ -730,6 +730,8 @@ class Test(Core, tmt.export.Exportable['Test']):
     duration: str = DEFAULT_TEST_DURATION_L1
     result: str = 'respect'
 
+    where: List[str] = []
+
     serialnumber: int = 0
```

Comment:
Just a quick brainstorming question: These two keys are kind-of "internal" as they are not part of the test specification. Shall we use `_where` instead to prevent possible future name clashes if new `Test` attributes are added to the spec? Similarly for the `serialnumber` key.

Not that I would expect exactly these two to be likely candidates for a clash, but if we're adding more and more...

Reply:
> Just a quick brainstorming question: These two keys are kind-of "internal" as they are not part of the test specification.

I disagree here WRT `serialnumber` - it's not part of the test specification, but it's part of *the results* specification, and as such is part of tmt's "public interface".

> Shall we use `_where` instead to prevent possible future name clashes if new `Test` attributes are added to the spec? 

Good question, but, in this particular case I'd say "not needed", and the following is the line of thoughts that got me there: ok, let's say we add a new key, conflicting with this `where` field. It would be called `where`, obviously, and what would it be for? There's already a `where` key defined, for the `prepare`, `execute`, and `finish` steps, to express the "run this phase on that guest/role" requirement. For the sake of consistency, I for one would expect the newly added key would play a very, very similar role in test specification, and that's exactly what the field added in this patch does. So in this particular case, and I can't be 100 % sure, of course, in my mind the risk of conflict is minimal.

> Similarly for the `serialnumber` key.

Different case, at least to me, as `serialnumber` has become part of the specification already. Of results, true, but this field has the very same meaning, it even **is** the same field, it's propagated from `Test` to `Result`, therefore it makes sense to me to use the same name.

Reply:
Makes sense. Thanks for sharing your thoughts.